### DISCUSSION TITLE
Question about MIDI Thru

### DISCUSSION THREAD

**Discussion by DutchBigEd** [probonopd/MiniDexed#513] [Q&A] [answered]
in the wiki it states:
Option MIDIThru=from,to in minidexed.ini is used forward MIDI events from the device from to to.

from/to can be:

umidiN (USB MIDI device, N = 1...4) (the MiniDexed device is acting as a USB host, not as a USB device)
ttyS1 (serial interface) (note that one can attach an [adapter](https://github.com/probonopd/MiniDexed/wiki/Hardware#usb-midi-device) that can act as a USB device which is attached to a tablet or computer)
ukbd1 (PC keyboard)

Can I use this? :
MIDIThru=ttyS1,ttyS1

**Comment by probonopd** [comment]
What would that be useful for? Have you tried it?

**Reply by DutchBigEd** [reply]
I have a limited number of hardware MIDI OUT DIN channels (5), So I can connect a MiniDexed DX7 to say MIDI OUT 1 acting on midi channel 1 and have another MiniDexed connected to the MIDI THRU of the first MiniDexed while acting on midi channels 2 - 9 
I have tried it (changed the minidexed.ini) but cant see a signal at midi thru.

**Reply by DutchBigEd** [reply, marked as answer]
ow.... it is working...
apparently the midi thru signal is too weak for the  minidexed midi input...
 sorry, you may remove this entry,...

**Reply by probonopd** [reply]
Thanks for the clarification and glad that it si working.

But what do you mean by

> the midi thru signal is too weak for the minidexed midi input...

**Reply by DutchBigEd** [reply]
current flowing thru the midi thru is too low, so I changed this:
![midi low](https://github.com/probonopd/MiniDexed/assets/17274454/e0a7079d-b5f1-42f1-872a-370c633bc331)

into this:
![midi ok](https://github.com/probonopd/MiniDexed/assets/17274454/c7501f20-9964-461c-abd9-315c227a1e32)
That solved the problem, but I forgot I had a jumper on my board to connect the midi thru directly to the midi in...

**Reply by probonopd** [reply]
Thanks for your explanation. Maybe someone can benefit from this in the future.

**Comment by diyelectromusic** [comment]
> apparently the midi thru signal is too weak for the minidexed midi input... sorry, you may remove this entry,...

What are you using as a MIDI OUTPUT circuit?  The MIDI Spec states it can be used with 3V3 logic systems by using a 33R and 10R resistors instead of the typical 220R used with 5V circuits.  This will ensure there is still there required 5mA current loop in operation to activate the receiver's optoisolator. More here: https://diyelectromusic.wordpress.com/2021/01/23/midi-micropython-and-the-raspberry-pi-pico/

But in general with a RPi I've found it advantageous to use a two of the inverters of a 74HCT14 buffer powered by 5V but driven by the Pi's 3V3 TX pin to create MIDI out. This is the approached used in the Clumsy MIDI interface and I've found it works well. It is also a good way to get a hardware THRU port from the RX side.  That is what I used in the HD44780 version of my own MiniDexed IO board, here: https://diyelectromusic.wordpress.com/2022/07/28/minidexed-raspberry-pi-io-board/

Kevin

**Reply by diyelectromusic** [reply]
Oh, and I should add that if you squint and the GNDs are connected, then actually the output of your 3V3 aware MIDI IN circuit could actually drive both RX pins on both Pis directly... but that is starting to stretch things a little... but it has worked for me in the past when I was being lazy! :)

Kevin

**Reply by DutchBigEd** [reply]
I used the schematics as in the previous reply.

**Reply by DutchBigEd** [reply]
Also, both minidexed dx7's suppose to be midi compatible to other midi devices, both input as wel as output wise.

**Comment by diyelectromusic** [comment]
> Also, both minidexed dx7's suppose to be midi compatible to other midi devices, both input as wel as output wise.

MiniDexed is fully software compatible with MIDI.  But the hardware one uses to provided a serial MIDI link is down to the individual.

There is a sample serial MIDI IN circuit on the Wiki, but no serial MIDI OUT or THRU, hence me asking what you used that led you to conclude that "the midi thru signal is too weak for the minidexed midi input".  From the circuit you've posted, you appear to have provided your own buffer stage and it looks like the fix you've implemented is to swap from 3V3 to 5V operation for your buffer?

As I said, it may well be that swapping your 220R resistors for 33R and 10R would also have solved your issue, as these are the recommended values to be used in an output stage for 3V3 operation whilst maintaining the required 5mA current loop for MIDI operation.

But as I say, whether MiniDexed is "midi compatible to other midi devices" is largely an exercise for the reader (at least hardware wise)! :)

Kevin

**Reply by DutchBigEd** [reply]
`As I said, it may well be that swapping your 220R resistors for 33R and 10R would also have solved your issue, as these are the recommended values to be used in an output stage for 3V3 operation whilst maintaining the required 5mA current loop for MIDI operation.`
Changing the 3V3 to 5V solved the problem and was far more easier as changing two resistors.
